python/tests/test_tree_positioning.py

```diff
@@ -135,9 +135,43 @@ def seek_forward(self, index):
                 # print("skip", j, e, self.ts.edges_left[e], self.ts.edges_right[e])
 
     def seek_backward(self, index):
-        # TODO
-        while self.tree_pos.index != index:
-            self.prev()
+        old_left, old_right = self.tree_pos.interval
+        old_index = self.tree_pos.index
+        self.tree_pos.seek_backward(index)
+        left, right = self.tree_pos.interval
+        # print("Current interval:", old_left, old_right)
+        # print("New interval:", left, right)
+        # print("index:", index, "out_range:", self.tree_pos.out_range)
+        if old_index == -1:
```

Comment:
Hmmm, this doesn't smell right to me. I think we need to look at the initial conditions to think about why we *don't* need to exclude for the null tree. I guess we should really take care of this by setting the out_range to be empty, in the TreePosition.seek_forward and TreePosition.seek_backward functions, rather than having to reason about it here. As you point out, we should have no edges to remove, so there's no point in iterating over the out_range at all.

Reply:
That makes sense to me, I'll give that a try now. It worked in the `seek_forward` case because it was checking if `e_left < old_right = 0` but I soon realised that it didn't work out so neatly when translated to `seek_backward`.

Reply:
After some experiments, an approach I've found that works is as follows, from the `TreePosition.seek_forward` function as an example:

```python
j = right_current_index
self.out_range.start = j
while j < M and right_coords[right_order[j]] <= left:
    j += 1
self.out_range.stop = j

if self.index == -1:
    #No edges, so out_range should be empty
    self.out_range.start = self.out_range.stop
```

The `out_range.stop` value still needs to be updated for later iterations to work, but setting start = stop prevents `StatefulTree.seek_forward` from iterating over the `out_range`. It works similarly in the `seek_backward` case. Is that what you have in mind?

Reply:
I just pushed a commit with these changes. The tests are rather trivial; I couldn't think of another way to check the `out_range` is empty without adding code to track if `StatefulTree.seek_x` is looping over the `out_range` or not.